### ISSUE TITLE
ActionView ap and ap_debug displays as string instead of HTML (with proposed patch)

### ISSUE THREAD

**Issue by JohnAtFenestra** [michaeldv/awesome_print#199]
Applies to:
```
awesome_print (1.6.1)
rails (4.2.1)
```

Adding the following for debugging purposes in a view:

```html
<%= ap params %>
```

displays the text as a string without any formatting. 

```
<pre class="debug_dump">{ &quot;controller&quot;<kbd style="color:slategray"> =&gt; </kbd><kbd style="color:brown">&quot;devise/sessions&quot;</kbd>, &quot;action&quot;<kbd style="color:slategray"> =&gt; </kbd><kbd style="color:brown">&quot;new&quot;</kbd> }</pre>
```

After researching the issue, I discovered the problem is in `lib/awesome_print/ext/action_view.rb`. If we make the following change to ap_debug, the view properly displays the output properly formatted as HTML.

```Ruby
    def ap_debug(object, options = {})
      # Addition of ::ActiveSupport::SafeBuffer.new by JE on 2015-07-21
      ::ActiveSupport::SafeBuffer.new object.ai(options.merge(:html => true)).sub(/^<pre([\s>])/, '<pre class="debug_dump"\\1')
    end
```

**Comment by IslamAzab** [comment]
I get something similar, any  luck with finding a fix for it ?

**Comment by gerrywastaken** [comment]
@IslamAzab: @JohnAtFenestra has already shown how it can be fixed but the change was never made to this library. I'm guessing this is because the change will only work with later versions of rails and might break for previous versions (just a guess).

It seems the author of awesome_print was creating version 2, but that hasn't been worked on since 2015, so things are now at the point where you or somebody else should probably apply the patch yourself to a fork.

**Comment by IslamAzab** [comment]
@gerrywastaken Thanks.

**Comment by flynfish** [comment]
Would be nice to include this in an updated version

**Comment by flynfish** [comment]
You can also just call html_safe on it:
```ruby
<%= (ap @model).html_safe %>
```

**Comment by gerrywastaken** [comment]
@flynfish There are 15 pull requests and the last commit was March of 2015. I'm guessing @michaeldv has more important things to work on. So I suspect it's at a stage of fork, or live with it.

**Comment by michaeldv** [comment]
@gerrywastaken It's not so simple as I work for the fruit company. I would be happy to provide commit/push rights to you or anyone interested. Let me know.

**Comment by gerrywastaken** [comment]
@michaeldv Makes sense. I wasn't complaining btw, I completely understand about having limited time. Having said that.... OK, if you give me rights I'll pitch in from time to time.

**Comment by IslamAzab** [comment]
@michaeldv Let me know what I can help with too, I am interested.

**Comment by michaeldv** [comment]
@IslamAzab Added you to https://github.com/michaeldv/awesome_print/settings/collaboration as well, please coordinate new version release with the folks there.

**Comment by gerrywastaken** [comment]
@michaeldv Thanks for adding me. I'm am unable to access that page and just get back a 404.

BTW, if anybody wants to submit a pull request for this issue, please go ahead. :)
